docs/docs/kb/13-debug.md

```diff
@@ -0,0 +1,204 @@
+description: Blockchain Debug
+---
+
+# Blockchain Debug
+
+Ignite chain debug command can help you to find issues during development.
+It uses [Delve](https://github.com/go-delve/delve) debugger which enables you to
+interact with your blockchain app by controlling the execution of the process,
+evaluating variables, and providing information of thread / goroutine state, CPU
+register state and more.
+
+## Debug Command
+
+The debug command requires that the blockchain app binary is build with
+debugging support by removing optimizations and inlining.
+A debug binary is built by default by the `ignite chain serve` command or can
+optionally be created using the `--debug` flag when running `ignite chain init`
+or `ignite chain build` sub-commands.
+
+To start a debugging session in the terminal run:
+
+```bash
+ignite chain debug
+```
+
+The command runs your blockchan app in the background, attaches to it and
+launches a terminal debugger shell:
+
+```
+Type 'help' for list of commands.
+(dlv)
+```
+
+At this point the blockchain app blocks execution so you can set one or more
+breakpoints before continuing execution.
+
+Use the [break](https://github.com/go-delve/delve/blob/master/Documentation/cli/README.md#break)
+(alias `b`) command to set any number of breakpoints using, for example the
+`<filename>:<line>` notation:
+
+```
+(dlv) break x/hello/client/cli/query_say_hello.go:14
+```
+
+This command adds a breakpoint to the `x/hello/client/cli/query_say_hello.go`
+file at line 14.
+
+Once all breakpoints are setted resume blockchain execution using the
+[continue](https://github.com/go-delve/delve/blob/master/Documentation/cli/README.md#continue)
+(alias `c`) command:
+
+```
+(dlv) continue
+```
+
+The debugger will launch the shell and stop blockchain execution again when a
+breakpoint is triggered.
+
+Within the debugger shell use the `quit` (alias `q`) or `exit` commands to stop
+the blockchain app and exit the debugger.
+
+## Debug Server
+
+A debug server can optionally be started in cases where the default terminal
+client is not desirable. When the server starts it first runs the blockchain
+app, attaches to it and finally waits for a client connection. It accepts both
+JSON-RPC or DAP client connections.
+
+To start a debug server use the following flag:
+
+```bash
+ignite chain debug --server
+```
+
+To start a debug server with a custom address use the following flags:
+
+```bash
+ignite chain debug --server --server-address 127.0.0.1:30500
+```
+
+The debug server stops automatically when the client connection is closed.
+
+## Debugging Clients
+
+### Gdlv: Multiplatform Delve UI
+
+[Gdlv](https://github.com/aarzilli/gdlv) is a graphical frontend to Delve for
+Linux, Windows and macOS.
+
+Using it as debugging client is straight forward as it doesn't require any
+configuration. Once the debug server is running and listening for client
+requests, for example at the default `tcp://127.0.0.1:30500` address, connect to
+it by running:
```

Comment:
I think the "for example at the default ... address" part should be moved earlier in the DEBUG SERVER section.  Then this sentence can be simplified.

Reply:
Changeset bddd289